disk/block_test.go

```diff
@@ -0,0 +1,151 @@
+package disk
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	tmpTestFile = "/tmp/cfs_disk_block_test"
+)
+
+var testFilePattern = []byte{'a', 'b', 'c', 'd', 'e', 'f', 'g'}
+
+func setUpTestFile(length int64, t *testing.T) *os.File {
+	f, err := os.OpenFile(tmpTestFile, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0600)
+	if err != nil {
+		t.Fatalf("can not open test file: %v", err)
+		return nil
+	}
+
+	// fill tmp test file with test pattern
+	for length > 0 {
+		if length < int64(len(testFilePattern)) {
+			_, err = f.Write(testFilePattern[0 : length-1])
+		} else {
+			_, err = f.Write(testFilePattern)
+		}
+		if err != nil {
+			t.Fatalf("can not write to test file: %v", err)
+		}
+		length = length - int64(len(testFilePattern))
+	}
+
+	if err = f.Sync(); nil != err {
+		t.Fatalf("can not sync test file: %v", err)
+	}
+	return f
+}
+
+func TestPartlyReadBlock(t *testing.T) {
+	f := setUpTestFile(90, t)
+	p := make([]byte, 6)
+	// write
+	err := writeBlock(f, 8, 10, p)
+	if err != nil {
+		t.Errorf("error = %v", err)
+	}
+	rp := make([]byte, 16)
+	n, err := readBlock(f, rp, 4, 20)
+	if err != io.EOF {
+		t.Errorf("expect error %v got %v", io.EOF, err)
+	}
+	if n != 6 {
+		t.Errorf("expect get data length = %v got %v", 6, n)
+	}
+}
+
+func TestBasicReadWriteBlock(t *testing.T) {
```

Comment:
why this is basic? do we expect to have a advanced one? I hope we can have a more concrete name for the test too.